libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/WayNameView.java

```diff
@@ -32,7 +33,20 @@ public WayNameView(@NonNull Context context, @Nullable AttributeSet attrs, int d
     initialize();
   }
 
+  @Override
+  public void setVisibility(int visibility) {
+    if (getVisibility() != visibility) {
+      super.setVisibility(visibility);
+    }
+  }
+
   public void updateWayNameText(String wayNameText) {
+    if (TextUtils.isEmpty(wayNameText)) {
```

Comment:
shouldn't check here if `updateVisibility(Boolean)` was called with `false` before?

Reply:
so my idea was to hide the view if setting an empty text no matter `updateVisibility(false)` was called before.